pkg/controller/logstash/volume/volumes.go

```diff
@@ -0,0 +1,114 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License 2.0;
+// you may not use this file except in compliance with the Elastic License 2.0.
+
+package volume
+
+import (
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+
+	commonv1 "github.com/elastic/cloud-on-k8s/v2/pkg/apis/common/v1"
+	logstashv1alpha1 "github.com/elastic/cloud-on-k8s/v2/pkg/apis/logstash/v1alpha1"
+	"github.com/elastic/cloud-on-k8s/v2/pkg/controller/common/volume"
+	"github.com/elastic/cloud-on-k8s/v2/pkg/utils/set"
+)
+
+// AppendDefaultPVCs appends defaults PVCs to a set of existing ones.
+//
+// The default PVCs are not appended if:
+// - a Volume with the same .Name is found in podSpec.Volumes, and that volume is not a PVC volume
+func AppendDefaultPVCs(existing []corev1.PersistentVolumeClaim, ls logstashv1alpha1.Logstash) []corev1.PersistentVolumeClaim {
```

Comment:
I think the word `existing` is descriptive in this context because we are mutating the given slice of (existing) PVCs. I propose to go with `existingPVCs`.